dashboard/src/pages/hardwareDetails/HardwareDetailsHeaderTable.tsx

```diff
@@ -98,46 +106,80 @@ const columns: ColumnDef<Trees>[] = [
   },
 ];
 
-const sanitizeTreeItems = (treeItems: Trees[]): Trees[] => {
-  return treeItems.map(tree => ({
-    treeName: tree['treeName'] ?? '-',
-    gitRepositoryBranch: tree['gitRepositoryBranch'] ?? '-',
-    headGitCommitName: tree['headGitCommitName'] ?? '-',
-    headGitCommitHash: tree['headGitCommitHash'],
-    gitRepositoryUrl: tree['gitRepositoryUrl'] ?? '-',
-    index: tree['index'],
-  }));
+const indexesFromRowSelection = (
+  rowSelection: RowSelectionState,
+  maxTreeItems: number,
+): number[] => {
+  const rowSelectionValues = Object.values(rowSelection);
+  if (
+    rowSelectionValues.length === maxTreeItems ||
+    rowSelectionValues.length === 0
+  ) {
+    return [];
+  } else {
+    return Object.keys(rowSelection).map(v => parseInt(v));
+  }
 };
 
-export function HardwareHeader({ treeItems }: IHardwareHeader): JSX.Element {
+export function HardwareHeader({
+  treeItems,
+  selectedIndexes = [],
+  updateTreeFilters,
+}: IHardwareHeader): JSX.Element {
   const [sorting, setSorting] = useState<SortingState>([
     { id: 'treeName', desc: false },
   ]);
   const [pagination, setPagination] = useState<PaginationState>({
     pageIndex: 0,
     pageSize: 5,
   });
-  // const [rowSelection, setRowSelection] = useState({});
 
-  const data = useMemo(() => {
-    return sanitizeTreeItems(treeItems);
-  }, [treeItems]);
+  const initialRowSelection = useMemo(() => {
+    if (selectedIndexes.length === 0) {
+      return Object.fromEntries(
+        Array.from({ length: treeItems.length }, (_, i) => [
+          i.toString(),
+          true,
+        ]),
+      );
+    } else {
+      return Object.fromEntries(
+        Array.from(selectedIndexes, treeIndex => [treeIndex.toString(), true]),
+      );
+    }
+  }, [selectedIndexes, treeItems]);
```

Comment:
- create a function outside the component for this logic
- nit: when you use `if(..) return` you don't need the `else` just `return`